### PR TITLE
feat(runtime): complete wave 3 lifecycle normalization

### PR DESCRIPTION
**Pull request by jamesryancooper**
## What
- promote the Wave 3 runtime contract family under `/.octon/framework/constitution/contracts/runtime/**`
- bind canonical run control and evidence roots before consequential side effects
- bridge mission-backed generated summaries and mission views onto a real retention-compliant run root
- refresh extension and capability publication state so broad runtime checks stay green

## Why
- Wave 3 requires run roots to become the primary execution-time unit of truth
- mission continuity and generated read models must consume run evidence rather than replace it
- the branch should carry the same validator-backed proof that the architecture proposal requires

## How
- added runtime-state, rollback-posture, checkpoint, replay-pointer, and retained-evidence contracts and wiring
- updated engine authorization and orchestration run tooling to materialize canonical lifecycle files
- seeded a retention-compliant bridge run (`run-wave3-runtime-bridge-20260327`) and matching decision (`dec-wave3-runtime-bridge-20260327`)
- republished extension/capability effective state and updated Wave 3 evidence records

## Profile Selection Receipt
- `release_state`: `pre-1.0`
- `change_profile`: `transitional`
- receipt: `/.octon/instance/cognition/context/shared/migrations/2026-03-27-runtime-lifecycle-normalization-cutover/plan.md`

## Implementation Plan
- promote the constitutional runtime family
- normalize runtime writers around canonical run control and evidence roots
- bridge mission continuity and generated read models to canonical run evidence
- verify completion through broad harness and mission-autonomy alignment checks

## Impact Map (code, tests, docs, contracts)
- code: runtime authorization/evidence wiring, orchestration run tooling, publication-state refresh
- tests: broad validator and alignment-profile coverage exercised in local verification
- docs: runtime, architecture, mission, and state docs now describe the Wave 3 model
- contracts: objective registry, runtime family, mission view schema, and validation/evidence records updated

## Compliance Receipt
- `cargo check --manifest-path .octon/framework/engine/runtime/crates/Cargo.toml`
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-runtime-lifecycle-normalization.sh`
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-continuity-memory.sh`
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-runtime-effective-state.sh`
- `bash .octon/framework/assurance/runtime/_ops/scripts/alignment-check.sh --profile harness`
- `bash .octon/framework/assurance/runtime/_ops/scripts/alignment-check.sh --profile mission-autonomy`

## Exceptions/Escalations
- `accept:human` is required because this change touches high-impact governance and runtime authority surfaces
- no further exceptions or escalations are required beyond the repo policy label

## Tradeoffs
- transitional compatibility surfaces remain where intended, but canonical readers now prefer the Wave 3 run-root families
- the branch commits publication-state refresh artifacts so the broad repo checks stay green

## Testing
- verified Wave 3 lifecycle validator, continuity retention contract, broad runtime-effective-state sweep, and both alignment profiles
- verified mission views consume a real retention-compliant run root and canonical receipt/checkpoint/replay artifacts

## Rollout
- merge as a normal PR after required review and green checks
- no separate deployment step; generated/runtime artifacts are repo-local

## Convivial Purpose Check
- this keeps the harness deterministic enough to trust, observable enough to debug, and flexible enough to evolve

## Risk Rubric
- risk level: moderate because the change touches runtime authority and generated mission consumers
- mitigations: canonical lifecycle contracts, broad validator coverage, retention-compliant bridge artifacts, and publication-state refresh

## Contracts and Threat Model
- canonical run roots now own execution-time lifecycle state and retained evidence
- host affordances remain projection-only; mission continuity remains a consumer, not the source of execution truth
- no new external network or secret-handling paths were introduced

## Observability and Performance
- run evidence now records canonical receipts, checkpoints, replay pointers, and retained evidence manifests
- publication-state refresh keeps generated runtime-effective surfaces aligned with the root manifest

## License and Provenance
- repo-local authored/governed changes only
- no external code import or license change in this branch
- No-Issue: repo-local Wave 3 constitutional runtime cutover

## Checklist
- [x] Requirements met; edge cases handled
- [x] Security reviewed (authz, input validation, secrets)
- [x] Tests added or updated
- [x] Observability updated (logs, metrics, traces) if needed
- [x] No speculative abstractions or unnecessary complexity
- [x] Conventions followed; no drift introduced
- [x] Non-obvious decisions documented (comments, ADR)
- [x] All review conversations resolved

### CODE REVIEW

File: .octon/framework/orchestration/runtime/_ops/scripts/write-run.sh [hunk 1]
```diff
@@ -245,6 +583,18 @@ update_run_contract_status() {
...
     mv "$initial_attempt.tmp" "$initial_attempt"
   fi
+
+  local mission_id parent_run_id
+  mission_id="$( [[ -f "$RUNTIME_RUNS_DIR/$run_id.yml" ]] && yq -r '.mission_id // ""' "$RUNTIME_RUNS_DIR/$run_id.yml" 2>/dev/null || printf '' )"
+  parent_run_id="$( [[ -f "$RUNTIME_RUNS_DIR/$run_id.yml" ]] && yq -r '.parent_run_id // ""' "$RUNTIME_RUNS_DIR/$run_id.yml" 2>/dev/null || printf '' )"
+  local summary=""
+  summary="$( [[ -f "$RUNTIME_RUNS_DIR/$run_id.yml" ]] && yq -r '.summary // ""' "$RUNTIME_RUNS_DIR/$run_id.yml" 2>/dev/null || printf '' )"
+  write_orchestration_receipt_file "$run_id" "$projection_status" "$summary" "$updated_at"
+  write_runtime_state_file "$run_id" "$projection_status" "${updated_at}" "$updated_at" "$mission_id" "$parent_run_id" "$(orchestration_receipt_relpath "$run_id")"
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .octon/framework/orchestration/runtime/_ops/scripts/write-run.sh]: Updated `write_runtime_state_file` to preserve the original `created_at` from an existing `runtime-state.yml` and to carry forward the last-known receipt/checkpoint refs when status updates do not replace them. Status transitions now only advance `updated_at`.

File: .octon/framework/orchestration/runtime/_ops/scripts/write-run.sh [hunk 2]
```diff
@@ -245,6 +583,18 @@ update_run_contract_status() {
...
+  write_rollback_posture_file "$run_id" "$updated_at"
+  write_replay_pointer_file "$run_id" "$updated_at"
+  write_trace_pointer_file "$run_id" "$updated_at"
+  write_retained_evidence_file "$run_id" "$updated_at"
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .octon/framework/orchestration/runtime/_ops/scripts/write-run.sh:680]: Updated `write-run.sh` so status updates merge existing replay, trace, and retained-evidence state instead of regenerating those manifests from orchestration defaults. Heartbeat, complete, and recovery now preserve runtime-appended receipt/checkpoint/trace links.
